File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/PostCustomRoleExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class PostCustomRoleExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey = new Dictionary<string, string> {["ApiKey"] = "YOUR_API_KEY"};

        var policy1 = new StatementPost(
            effect: StatementPost.EffectEnum.Allow,
            resources: [
                "proj/*:env/production:flag/*",
            ],
            actions: [
                "updateOn",
            ]
        );

        var policy = new List<StatementPost>
        {
            policy1,
        };

        var customRolePost = new CustomRolePost(
            name: "Ops team",
            key: "role-key-123abc",
            description: "An example role for members of the ops team",
            basePermissions: "reader",
            policy: policy
        );

        try
        {
            var response = new CustomRolesApi(config).PostCustomRole(
                customRolePost: customRolePost
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling CustomRoles#PostCustomRole: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/PostCustomRoleExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const policy1: api.StatementPost = {
  effect: api.StatementPostEffectEnum.Allow,
  resources: [
    "proj/*:env/production:flag/*",
  ],
  actions: [
    "updateOn",
  ],
};

const policy = [
  policy1,
];

const customRolePost: api.CustomRolePost = {
  name: "Ops team",
  key: "role-key-123abc",
  description: "An example role for members of the ops team",
  basePermissions: "reader",
  policy: policy,
};

new api.CustomRolesApi(configuration).postCustomRole(
  customRolePost,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling CustomRolesApi#postCustomRole:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
